NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -330,25 +330,21 @@ public boolean hasAccess(Group group, UUID player, PermissionType perm) {
 			}
 		}
 		GroupPermission perms = getPermissionforGroup(group);
-		for(PlayerType rank : getRecursivePlayerTypes(group, player)) {
-			if (perms.hasPermission(rank, perm)) {
-				//player has right rank in the group itself or at least one super group
-				return true;
-			}
-		}		
+		PlayerType rank = getRecursiveHighestPlayerType(group, player); 
+		if (perms.hasPermission(rank, perm)) {
+			//player has right rank in the group itself or at least one super group
+			return true;
+		}	
 		return false;
 	}
 	
-	private List<PlayerType> getRecursivePlayerTypes(Group group, UUID player) {
-		List<PlayerType> perms = new LinkedList<PlayerType>();
+	private PlayerType getRecursiveHighestPlayerType(Group group, UUID player) {
 		PlayerType type = group.getPlayerType(player);
-		if (type != null) {
-			perms.add(type);
-		}
 		if (group.hasSuperGroup()) {
-			perms.addAll(getRecursivePlayerTypes(group.getSuperGroup(), player));
+			//only return highest rank
+			type = PlayerType.getByID(Math.max(PlayerType.getID(type), PlayerType.getID(getRecursiveHighestPlayerType(group.getSuperGroup(), player))));
 		}
-		return perms;
+		return type;
```

Comment:
Hmm, there are a few weird edge cases with this where it's advantageous to create dummy super groups, just so for example your mods also get special permissions only members have. Trying to support those weirdly structured groups leads to a lot of things that dont really make sense and shouldnt be possible, but whatever. If you think the current system is better, I'll revert it.